src/script/lua_api/l_object.cpp

```diff
@@ -1803,6 +1803,11 @@ int ObjectRef::l_set_sky(lua_State *L)
 			// pop "sky_color" table
 			lua_pop(L, 1);
 		}
+		lua_getfield(L, 2, "fog");
+		if (lua_istable(L, -1)) {
+			sky_params.fog_distance = getfloatfield_default(L, -1,  "fog_distance", -1);
+			sky_params.fog_start = getfloatfield_default(L, -1,  "fog_start", -1);
```

Comment:
Or, so `{fog = {}}` resets all fog params?
Wouldn't it be better to make it the same as the params in the sky params table, i.e. don't set if nil?

Reply:
Above: I prefer the if. It's not an exceptional case, but something we expect. In other parts we also use this if-statement.

Reply:
Re: `fog={}` good catch. Same with `fog={fog_distance=X}`, which would unset the fog_start value which is should not.